src/model/FormGenerator.js

```diff
@@ -30,31 +30,36 @@ export default class FormGenerator {
    * @return Promise with generated form definition and form data
    */
   static constructForm(structure, intl) {
-    return new Promise((resolve) =>
-      jsonld.flatten(structure, {}, null, (err, structure) => {
-        let formProperties;
-        let form;
-        if (err) {
-          Logger.error(err);
-        }
+    let formProperties;
+    let form;
+    return jsonld
+      .flatten(structure, {})
+      .then((flattenedStructure) => {
         try {
           const [formQuestions, rootForm] =
-            FormGenerator._constructFormQuestions(structure, intl);
+            FormGenerator._constructFormQuestions(flattenedStructure, intl);
           form = rootForm;
           formProperties = {
             formQuestions,
           };
         } catch (e) {
+          console.error("Error in _constructFormQuestions:", e);
```

Comment:
I suggest to have some meaningful message here instead of pointing to private method `_constructFormQuestions`

Reply:
I updated the message.